src/huggingface_hub/hf_api.py

```diff
@@ -1482,8 +1510,10 @@ def upload_file(
         d = r.json()
         return d["url"]
 
+    @_deprecate_positional_args
     def delete_file(
         self,
+        *,
         path_in_repo: str,
         repo_id: str,
```

Comment:
Here too, I'd put it after the `path_in_repo`, personally
```suggestion
    @_deprecate_positional_args
    def delete_file(
        self,
        path_in_repo: str,
        *,
        repo_id: str,
```